File: 80.remove-duplicates-from-sorted-array-ii.py
class Solution:
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums) <= 1:
            return len(nums)
        i = 1
        for j in nums[2:]:
            if j != nums[i - 1]:
                i += 1
                nums[i] = j
        return i + 1

File: 80.remove-duplicates-from-sorted-array-ii.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.size() <= 1) {
            return nums.size();
        }
        int i = 1;
        for (int j = 2; j < nums.size(); j++) {
            if (nums[j] != nums[i - 1]) {
                i++;
                nums[i] = nums[j];
            }
        }
        return i + 1;
    }
};